From c68162d14e3952bb398aa72105e0f666d2ba1169 Mon Sep 17 00:00:00 2001
From: Jacob Chen <jacob-chen@iotwrt.com>
Date: Thu, 27 Apr 2017 00:02:27 +0800
Subject: [PATCH] fix access violation in system compatible reading

This bug will casue
"ptr = soc_name + first compatible length + all compatible length"
and what we need is
"ptr = soc_name + first compatible length + second compatible length""

Signed-off-by: Jacob Chen <jacob-chen@iotwrt.com>
---
 osal/mpp_platform.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/osal/mpp_platform.cpp b/osal/mpp_platform.cpp
index 4c0f50f..bef415a 100644
--- a/osal/mpp_platform.cpp
+++ b/osal/mpp_platform.cpp
@@ -175,7 +175,7 @@ MppPlatformService::MppPlatformService()
                 char *ptr = soc_name;
 
                 while (ptr < soc_name + soc_name_len - 1) {
-                    ptr += strnlen (soc_name, MAX_SOC_NAME_LENGTH);
+                    ptr += strnlen (ptr, MAX_SOC_NAME_LENGTH);
                     *ptr = ' ';
                 }
 
-- 
2.7.4

